lib/websocket-tracker.js

```diff
@@ -32,7 +32,10 @@ function WebSocketTracker (client, announceUrl, opts) {
   self._interval = null
 
   if (socketPool[announceUrl]) self._socket = socketPool[announceUrl]
-  else self._socket = socketPool[announceUrl] = new Socket(announceUrl)
+  else {
+      self._socket = socketPool[announceUrl] = new Socket(announceUrl)
+      self._socket.on('ready', function () { self._socket.ready = true })
```

Comment:
Hmm, is that version released. I only just installed this and the version pulled down didn't have that.